scalameter-core/src/main/scala/org/scalameter/execution/invocation/instrumentation/Instrumentation.scala

```diff
@@ -0,0 +1,107 @@
+package org.scalameter.execution.invocation.instrumentation
+
+import java.io._
+import java.util.jar.{JarEntry, JarOutputStream, Attributes}
+import java.util.zip.ZipFile
+import org.objectweb.asm.{ClassReader, ClassWriter}
+import scala.collection.convert.decorateAsScala._
+import org.scalameter._
+import org.scalameter.execution.invocation._
+
+
+private[scalameter] object Instrumentation {
+  /** Reads classes from given location and transforms them if they match a predicate.
+   *
+   *  Note that it supports reading from directories and jars.
+   *
+   *  @param in location of classes to read from
+   *  @param p matches if class should be instrumented
+   *  @param instrumentFrom instruments class from a [[java.io.InputStream]]
+   *  @return [[scala.Iterator]] of tuples in a form of (fullClassName, instrumentedClassBytes)
+   */
+  private def transformClasses(in: File)(p: String => Boolean)(instrumentFrom: (String, InputStream) => Array[Byte]): Iterator[(String, Array[Byte])] = {
+    val dirFilter = new FilenameFilter {
+      def accept(file: File, s: String): Boolean = new File(file, s).isDirectory
+    }
+
+    val classFilter = new FilenameFilter {
+      def accept(file: File, s: String): Boolean = s.endsWith(".class")
+    }
+
+    def _transformClasses(in: File, rootPath: String): Iterator[(String, Array[Byte])] = {
+      val subdirs = in.listFiles(dirFilter)
+      val classes = in.listFiles(classFilter)
+      classes.iterator.collect { case f if p(s"$rootPath/${f.getName}".stripPrefix("/").stripSuffix(".class").replace('/', '.')) =>
+        val path = s"$rootPath/${f.getName}".stripPrefix("/")
+        path -> instrumentFrom(path.stripSuffix(".class"), new FileInputStream(f))
+      } ++ subdirs.iterator.flatMap(f => _transformClasses(f, s"$rootPath/${f.getName}".stripPrefix("/")))
+    }
+
+    if (in.isDirectory) {
+      _transformClasses(in, rootPath = "")
+    } else { // if it's not directory try to read it as zip/jar file
+      val zipFile = new ZipFile(in)
+      zipFile.entries().asScala.collect {
+        case entry if p(entry.getName.stripSuffix(".class").replace('/', '.')) =>
+          entry.getName -> instrumentFrom(entry.getName.stripSuffix(".class"), zipFile.getInputStream(entry))
```

Comment:
Please break this line so that it does not cross the 88 character limit:

    entry.getName -> instrumentFrom(
      entry.getName.stripSuffix(".class"), ...)